modules/core/shared/src/main/scala/scalacache/Entry.scala

```diff
@@ -1,16 +1,31 @@
 package scalacache
 
-import java.time.{Clock, Instant}
+import java.time.Instant
+import cats.effect.Clock
+import java.util.concurrent.TimeUnit
+import cats.implicits._
+import language.higherKinds
+import cats.Applicative
 
 /**
   * A cache entry with an optional expiry time
   */
-case class Entry[+A](value: A, expiresAt: Option[Instant]) {
+case class Entry[+A](value: A, expiresAt: Option[Instant])
+
+object Entry {
 
   /**
     * Has the entry expired yet?
     */
-  def isExpired(implicit clock: Clock): Boolean =
-    expiresAt.exists(_.isBefore(Instant.now(clock)))
+  def isExpired[F[_], A](entry: Entry[A])(implicit clock: Clock[F], applicative: Applicative[F]): F[Boolean] =
+    entry.expiresAt
+      .traverse { expiration =>
+        val now = clock.monotonic(TimeUnit.MILLISECONDS).map(Instant.ofEpochMilli(_))
 
+        now.map(expiration.isBefore(_))
+      }
+      .map {
+        case None | Some(true) => true
+        case Some(false)       => false
+      }
```

Comment:
What about `.forall(identity)`?
Maybe that would just be harder to read, but thought I'd throw it out there.

Reply:
Yeah, I had that but decided an explicit match would be more readable (in fact, I had a bug here due to having used `contains` / `exists`).

Reply:
Cool, yeah I agree that the match is more readable. You don't have to do the mental gymnastics of remembering `forall` defaults to true.